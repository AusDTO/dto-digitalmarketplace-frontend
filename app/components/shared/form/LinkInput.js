import React from 'react'
import times from 'lodash/times'
import Textfield from './Textfield'
import './scss/LinkInput.scss'

import { validLink } from '../../validators'

class LinkInput extends React.Component {
  static defaultProps = {
    mapProps: {}
  }

  static propTypes = {
    id: React.PropTypes.string.isRequired,
    name: React.PropTypes.string.isRequired,
    htmlFor: React.PropTypes.string.isRequired,
    label: React.PropTypes.string.isRequired,
    model: React.PropTypes.oneOfType([
      React.PropTypes.func,
      React.PropTypes.string
    ]).isRequired,

    messages: React.PropTypes.object,
    validators: React.PropTypes.object,
    description: React.PropTypes.string,
    controlProps: React.PropTypes.object,
    mapProps: React.PropTypes.oneOfType([
      React.PropTypes.func,
      React.PropTypes.object
    ])
  }

  constructor(props) {
    super(props)

    this.state = { length: props.defaultRows || 1 }
  }

  addRow(e: ReactEvent) {
    e.preventDefault()

    this.setState({
      length: this.state.length + 1
    })
  }

  removeRow(id: number, e: ReactEvent) {
    e.preventDefault()

    this.setState({
      length: this.state.length > 1 ? this.state.length - 1 : 1
    })
  }

  render() {
    const { id, htmlFor, label, description, model } = this.props

    return (
      <div className="field" id={id}>
        <h3 style={{ marginTop: 0 }}>
          <label htmlFor={htmlFor}>
            {label}
          </label>
        </h3>
        {description &&
          <p className="hint" id={`${id}-hint`}>
            {description}
          </p>}
        <div>
          {times(this.state.length, i => {
            return (
              <span key={id + i}>
                {i > 0 &&
                  <span>
                    <hr styleName="hr" />

                    <div className="row">
                      <div className="col-xs-8 col-sm-10">
                        <h3 style={{ marginTop: 0, marginBottom: 0 }}>
                          Additional link (optional)
                        </h3>
                      </div>
                      <div className="col-xs-4 col-sm-2">
                        <button
                          type="button"
                          className="button-secondary col-xs-12 col-sm-3"
                          styleName="remove-button"
                          onClick={this.removeRow.bind(this, i)}
                        >
                          Remove{' '}
                          <span className="visuallyhidden">number {i + 1}</span>
                        </button>
                      </div>
                    </div>
                  </span>}
                <div className="row">
                  <div className="col-xs-12 col-sm-10">
                    <Textfield
                      model={`${model}.${i}.title`}
                      name="title"
                      id="title"
                      htmlFor="title"
                      label="Link Title"
                      description="As you would like it shown on the Digital Marketplace."
                    />

                    <Textfield
                      model={`${model}.${i}.url`}
                      label="Link URL"
                      name="url"
                      id="url"
                      htmlFor="url"
                      validators={{ validLink }}
                      messages={{
                        validLink: 'Links provided must begin with http'
                      }}
                    />
                  </div>
                </div>
              </span>
            )
          })}
          <button
            type="button"
            className="button-secondary"
            styleName="anotherLinkButton"
            onClick={this.addRow.bind(this)}
          >
            Add another link
          </button>
        </div>
      </div>
    )
  }
}
export default LinkInput
