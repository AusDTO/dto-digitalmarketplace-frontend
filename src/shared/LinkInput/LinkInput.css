.heading {
    margin:0;
}

.remove-button {
    float:right;
    margin: 0;
}

.hr {
    margin-top: 24px;
    margin-bottom: 37px;
}

.footer {
    margin:0;
}
