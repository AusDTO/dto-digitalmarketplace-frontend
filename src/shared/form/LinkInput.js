import React from 'react';
import { Control, controls } from 'react-redux-form';

import StatefulError from './StatefulError';
import LinkInputComponent from '../LinkInput';

const LinkInput = (props) => {
  const {
    id,
    name,
    htmlFor,
    label,
    description,
    model,
    messages,
    controlProps,
    mapProps,
    validators
  } = props;

  return (
    <div className="field" id={id}>
      <h3 style={{marginTop: 0}}><label htmlFor={htmlFor}>{label}</label></h3>
      {description && (
        <p className="hint" id={`${id}-hint`}>{description}</p>
      )}
      <Control
        model={model}
        controlProps={{ name, id, describedby: `${id}-hint`, hint: description, ...controlProps}}
        mapProps={{
          rows: ({ viewValue }) => viewValue,
          className: ({ fieldValue }) => !fieldValue.valid && fieldValue.touched ? 'invalid' : '',
          errors:
          (messages && <StatefulError
              model={model}
              messages={messages}
              id={id}
          />),
          ...mapProps,
          ...controls.default,
        }}
        validators={validators}
        component={LinkInputComponent}
      />


    </div>
  );
};

LinkInput.defaultProps = {
  mapProps: {},
};

LinkInput.propTypes = {
  id: React.PropTypes.string.isRequired,
  name: React.PropTypes.string.isRequired,
  htmlFor: React.PropTypes.string.isRequired,
  label: React.PropTypes.string.isRequired,
  model: React.PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string,
  ]).isRequired,

  messages: React.PropTypes.object,
  validators: React.PropTypes.object,
  description: React.PropTypes.string,
  controlProps: React.PropTypes.object,
  mapProps: React.PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.object,
  ]),
}

export default LinkInput;