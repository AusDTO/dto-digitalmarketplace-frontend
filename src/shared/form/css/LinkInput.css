.heading {
    margin:0;
}

.remove-button {
    float:right;
    margin: 0;
}

.hr {
    margin-bottom: 24px;
}

.footer {
    margin:0;
}

.anotherLinkButton {
    margin-top: 0;
}