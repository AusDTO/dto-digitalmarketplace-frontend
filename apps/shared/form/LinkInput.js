/* eslint-disable react/jsx-no-bind */
import React from 'react'
import PropTypes from 'prop-types'
import times from 'lodash/times'
import Textfield from './Textfield'
import './scss/LinkInput.scss'

import { validLink } from '../../validators'

class LinkInput extends React.Component {
  constructor(props) {
    super(props)

    this.state = { length: props.defaultRows || 1 }
  }

  addRow(e: ReactEvent) {
    e.preventDefault()

    this.setState({
      length: this.state.length + 1
    })
  }

  removeRow(id: number, e: ReactEvent) {
    e.preventDefault()

    this.setState({
      length: this.state.length > 1 ? this.state.length - 1 : 1
    })
  }

  render() {
    const { id, htmlFor, label, description, model } = this.props
    return (
      <div className="field" id={id}>
        <h3>
          <label htmlFor={htmlFor}>
            {label}
          </label>
        </h3>
        {description &&
          <p className="hint" id={`${id}-hint`}>
            {description}
          </p>}
        <div>
          {times(this.state.length, i =>
            <span key={id + i}>
              {i > 0 &&
                <span>
                  <hr styleName="hr" />

                  <div className="row">
                    <div className="col-xs-8 col-sm-10">
                      <h3>Additional link (optional)</h3>
                    </div>
                    <div className="col-xs-4 col-sm-2">
                      <button
                        type="button"
                        className="button-secondary col-xs-12 col-sm-3"
                        styleName="remove-button"
                        onClick={this.removeRow.bind(this, i)}
                      >
                        Remove <span className="visuallyhidden">number {i + 1}</span>
                      </button>
                    </div>
                  </div>
                </span>}
              <div className="row">
                <div className="col-xs-12 col-sm-10">
                  <Textfield
                    model={`${model}.${i}.title`}
                    name="title"
                    id="title"
                    htmlFor="title"
                    label="Link Title"
                    description="As you would like it shown on the Digital Marketplace."
                  />

                  <Textfield
                    model={`${model}.${i}.url`}
                    label="Link URL"
                    name="url"
                    id="url"
                    htmlFor="url"
                    validators={{ validLink }}
                    messages={{
                      validLink: 'Links provided must begin with http'
                    }}
                  />
                </div>
              </div>
            </span>
          )}
          <button
            type="button"
            className="button-secondary"
            styleName="anotherLinkButton"
            onClick={this.addRow.bind(this)}
          >
            Add another link
          </button>
        </div>
      </div>
    )
  }
}

LinkInput.defaultProps = {
  description: ''
}

LinkInput.propTypes = {
  id: PropTypes.string.isRequired,
  htmlFor: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  model: PropTypes.oneOfType([PropTypes.func, PropTypes.string]).isRequired,
  description: PropTypes.string
}

export default LinkInput
