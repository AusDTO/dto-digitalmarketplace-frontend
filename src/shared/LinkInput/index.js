import LinkInput from './LinkInput'

export { LinkInput as default, LinkInput }