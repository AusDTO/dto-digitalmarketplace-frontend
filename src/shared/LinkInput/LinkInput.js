import React from 'react'
import {validLink} from '../../validators'

import './LinkInput.css'

class LinkInput extends React.Component {
    static propTypes = {
        rows: React.PropTypes.array,
        defaultRows: React.PropTypes.number,
        onChange: React.PropTypes.func,
        name: React.PropTypes.string.isRequired
    }

    static defaultProps = {
        onChange: () => {
        },
        onBlur: () => {
        },
        onFocus: () => {
        },
    }

    constructor(props) {
        super(props);

        let inputs;
        if (props.rows && props.rows.length) {
            // handle legacy data
            if (typeof props.rows[0] === 'string') {
                inputs = props.rows.map(value => {
                    return {url: value}
                });
                inputs = inputs.map((row, i) => this.createRow(i, row));
            } else {
                inputs = props.rows.map((row, i) => this.createRow(i, row))
            }
        }

        this.state = {
            inputs: inputs || Array
                .from({length: props.defaultRows || 1})
                .map((_, i) => this.createRow(i))
        };
    }

    /**
     * Pad out new row
     * @param  {number}
     * @param  {String}
     * @return {object}
     */
    createRow(id: number, value: string = '') {
        return {id, value}
    }

    /**
     * Get an array of values from the inputs.
     * Removes empty values
     * @return {array}
     */
    getValues() {
        return this.state.inputs.map(input => input.value).filter(i => i);
    }

    onChange(id: number, e: ReactEvent) {
        const {onChange} = this.props;

        // @see https://fb.me/react-event-pooling
        e.persist();

        this.setState((previousState) => {
            let newInputs = previousState.inputs.map(input => {
                if (input.id === id) {
                    input.value = {}
                    if (e.target.parentNode) {
                        Object.values(e.target.parentNode.getElementsByTagName('input')).forEach(field => {
                            input.value[field.getAttribute('data-field')] = field.value
                        })
                    }
                }

                return input;
            });

            return {inputs: newInputs};
        }, () => onChange(this.getValues()));

    }

    addRow(e: ReactEvent) {
        e.preventDefault();
        const {inputs} = this.state;
        const lastRow = inputs[inputs.length - 1];

        this.setState({
            inputs: inputs.concat(this.createRow(lastRow.id + 1))
        });

    }

    removeRow(id: number, e: ReactEvent) {
        e.preventDefault()
        const {inputs} = this.state;
        const {onChange} = this.props;

        this.setState({
            inputs: inputs.filter(r => r.id !== id)
        }, () => onChange(this.getValues()));
    }

    render() {
        const {inputs} = this.state;
        const {name, onBlur, onFocus, className = '', describedby, hint, errors} = this.props;
        return (
            <div>
                {inputs.map(({id, value}, i) => {
                    return (
                        <span key={id}>
                            {i > 0 && <span><hr styleName="hr"/>

                                <div className="row">
                                        <div className="col-xs-8 col-sm-10">
                                          <h3 style={{marginTop: 0, marginBottom:0}}>Additional link (optional)</h3>
                                        </div>
                                        <div className="col-xs-4 col-sm-2">
                                             <button type="button" className="button-secondary col-xs-12 col-sm-3"
                                                     styleName="remove-button"
                                                     onClick={this.removeRow.bind(this, id)}>
                                        Remove <span className="visuallyhidden">number {i + 1}</span>
                                    </button>
                                        </div>
                                    </div>
                            </span>}
                            <div className="row">
                                        <div className="col-xs-12 col-sm-10">
                <div className="field">
                  <label htmlFor={`${name}-title-${i}`} className="question-heading">Link Title</label>
                    <input
                        type="text"
                        name={`${name}-title[]`}
                        id={`${name}-title-${i}`}
                        data-field="title"
                        onChange={this.onChange.bind(this, id)}
                        onBlur={onBlur}
                        onFocus={onFocus}
                        className={`text-box ${className}`}
                        aria-describedby={hint && describedby}
                        defaultValue={value.title}
                        style={{outline: "0"}}/>
                </div>

                <div className="field">
                    <label htmlFor={`${name}-url-${i}`} className="question-heading">Link URL</label>
                    {!validLink(this.state.inputs[i].value.url) && errors}
                    <input
                        type="text"
                        name={`${name}-url[]`}
                        id={`${name}-url-${i}`}
                        data-field="url"
                        onChange={this.onChange.bind(this, id)}
                        onBlur={onBlur}
                        onFocus={onFocus}
                        className={`text-box ${className}`}
                        aria-describedby={hint && describedby}
                        defaultValue={value.url}
                        style={ (!validLink(this.state.inputs[i].value.url) ? {}: {outline:"0"}) }/>
                 </div>
              </div>

            </div>
              </span>
                    )
                })}
                <button type="button" className="button-secondary" styleName="anotherLinkButton" onClick={this.addRow.bind(this)}>
                    Add another link
                </button>
            </div>
        )
    }
}

export default LinkInput;