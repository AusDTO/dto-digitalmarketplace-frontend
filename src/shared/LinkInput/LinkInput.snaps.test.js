import React from 'react'
import renderer from 'react-test-renderer'

import LinkInput from './LinkInput'

test('LinkInput with no props', () => {
  const component = renderer.create(
    <LinkInput name="test" />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('LinkInput with 4 rows', () => {
  const rows = [ 'One', 'Two', 'Three', 'Four' ];
  const component = renderer.create(
    <LinkInput name="test" rows={rows} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('LinkInput with 4 default rows', () => {
  const component = renderer.create(
    <LinkInput name="test" defaultRows={4} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
