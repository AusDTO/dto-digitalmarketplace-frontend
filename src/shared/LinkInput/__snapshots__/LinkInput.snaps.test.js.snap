// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkInput with 4 default rows 1`] = `
<div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        1
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-0"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-0"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-0"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue={undefined}
        id="test-url-0"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        2
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-1"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-1"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-1"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue={undefined}
        id="test-url-1"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        2
      </span>
    </button>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        3
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-2"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-2"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-2"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue={undefined}
        id="test-url-2"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        3
      </span>
    </button>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        4
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-3"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-3"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-3"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue={undefined}
        id="test-url-3"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        4
      </span>
    </button>
  </div>
  <button
    className="button-secondary"
    onClick={[Function]}
    type="button"
  >
    Add another row
  </button>
</div>
`;

exports[`LinkInput with 4 rows 1`] = `
<div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        1
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-0"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-0"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-0"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue="One"
        id="test-url-0"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        2
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-1"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-1"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-1"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue="Two"
        id="test-url-1"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        2
      </span>
    </button>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        3
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-2"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-2"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-2"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue="Three"
        id="test-url-2"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        3
      </span>
    </button>
  </div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        4
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-3"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-3"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-3"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue="Four"
        id="test-url-3"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
    <button
      className="button-secondary col-xs-12 col-sm-3"
      onClick={[Function]}
      type="button"
    >
      remove 
      <span
        className="visuallyhidden"
      >
        number 
        4
      </span>
    </button>
  </div>
  <button
    className="button-secondary"
    onClick={[Function]}
    type="button"
  >
    Add another row
  </button>
</div>
`;

exports[`LinkInput with no props 1`] = `
<div>
  <div
    className="link-entry list-entry row"
  >
    <div
      className="col-xs-12 col-sm-9"
    >
      <strong>
        1
        .
      </strong>
      <br />
      <label
        htmlFor="test-title-0"
      >
         Title:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="title"
        defaultValue={undefined}
        id="test-title-0"
        name="test-title[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
      <label
        htmlFor="test-url-0"
      >
        URL:
      </label>
      <input
        aria-describedby={undefined}
        className="text-box "
        data-field="url"
        defaultValue={undefined}
        id="test-url-0"
        name="test-url[]"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
  <button
    className="button-secondary"
    onClick={[Function]}
    type="button"
  >
    Add another row
  </button>
</div>
`;
